Stop mapping zigbee switches to lights & switches.

igorbernstein2 opened the pull request:
## Description:
Zigbee switches only contain client clusters that are meant to control server clusters on a different device/endpoint.
This device type -> cluster mapping can be found in https://www.nxp.com/docs/en/user-guide/JN-UG-3076.pdf.
The intended usage is of connecting client clusters to server clusters is described in https://products.currentbyge.com/sites/products.currentbyge.com/files/document_file/DT200-GE-Zigbee-Primer-Whitepaper.pdf

The lack of server clusters on switches has been verified using a GE ZigBee Lighting Switch 45856GE and a 45857GE dimmer.

Output from a 45856GE:
```
Device:
  NWK: 0x0cd8
  IEEE: 00:22:a3:00:00:1f:37:68
  Endpoints:
    1: profile=0x104, device_type=DeviceType.ON_OFF_LIGHT
      Input Clusters:
        Basic (0)
        Identify (3)
        Groups (4)
        Scenes (5)
        On/Off (6)
        Metering (1794)
        Diagnostic (2821)
      Output Clusters:
        Time (10)
        Ota (25)
    2: profile=0x104, device_type=DeviceType.ON_OFF_LIGHT_SWITCH
      Input Clusters:
        Basic (0)
        Identify (3)
        Diagnostic (2821)
      Output Clusters:
        Identify (3)
        On/Off (6)
```

**Related issue (if applicable):** fixes #<home-assistant issue number goes here>
fixes #11847

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#<home-assistant.github.io PR number goes here>

## Example entry for `configuration.yaml` (if applicable):
```yaml

```

## Checklist:
  - [x] The code change is tested and works locally.

If user exposed functionality or configuration variables are added/changed:
  - [N/A] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [N/A] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [N/A] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [N/A] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [N/A] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [N/A] Tests have been added to verify that the new code works.

homeassistant commented:
Hi @igorbernstein, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#12280).

Once you do that we will be able to review and accept this pull request.

Thanks!

igorbernstein2 commented:
cla signed

igorbernstein2 committed:
Stop mapping zigbee switches to lights & switches.

Zigbee switches only contain client clusters that are meant to control server clusters on a different device/endpoint.
This device type -> cluster mapping can be found in https://www.nxp.com/docs/en/user-guide/JN-UG-3076.pdf.
The intended usage is of connecting client clusters to server clusters is described in https://products.currentbyge.com/sites/products.currentbyge.com/files/document_file/DT200-GE-Zigbee-Primer-Whitepaper.pdf

The lack of server clusters on switches has been verified using a GE ZigBee Lighting Switch 45856GE and a 45857GE dimmer.

Output from a 45856GE:
Device:
  NWK: 0x0cd8
  IEEE: 00:22:a3:00:00:1f:37:68
  Endpoints:
    1: profile=0x104, device_type=DeviceType.ON_OFF_LIGHT
      Input Clusters:
        Basic (0)
        Identify (3)
        Groups (4)
        Scenes (5)
        On/Off (6)
        Metering (1794)
        Diagnostic (2821)
      Output Clusters:
        Time (10)
        Ota (25)
    2: profile=0x104, device_type=DeviceType.ON_OFF_LIGHT_SWITCH
      Input Clusters:
        Basic (0)
        Identify (3)
        Diagnostic (2821)
      Output Clusters:
        Identify (3)
        On/Off (6)

balloob commented:
@rcloran , what do you think?

rcloran reviewed (approved):
Looks good to me

tinloaf commented:
Isn't this a breaking change? It will make existing entities switch from the `switch` domain to the `light` domain, right? That changes their entity ID.

igorbernstein2 commented:
From my experience, the switch device types would fail to be detected properly. Zha would discover the device type and try to load the load platform switch/zha or light/zha. This would fail because the discovered device would not have the necessary input/server clusters. So it would fail to add HA.

I think there are 2 possibilities:
1. The switch device is similar to my device. The result would be that the user's logs would no longer have an error
2. A physical light switch is labelled as a switch device type with additional on/off clusters. Which would be a breaking change. The user will have to explicitly override the component to light/switch. I don't have any data.

I only have anecdotal data from my device  and another user that chimed in https://github.com/home-assistant/home-assistant/issues/11847#issuecomment-363293175.

I have a suspicion that these device types are fairly rare.

rcloran commented:
Yeah, this may break some weird devices, but including these profiles was incorrect. Endpoints of these types don’t act as lights or switches that we can control. I think we should just do this — with #12331 people can mangle entity IDs themselves if they want. 

If you do want to add a release not about a breaking change (and I don’t think it’s necessary), I suggest something like: Switch or light devices which do not properly conform to the ZHA specification may change entity IDs after update
